h/static/scripts/upgrade-elements.js

```diff
@@ -0,0 +1,29 @@
+'use strict';
+
+/**
+ * Upgrade elements on the page with additional functionality
+ *
+ * @param {Element} root - The root element to search for matching elements
+ * @param {Object} controllers - Object mapping selectors to controller classes.
+ *        For each element matching a given selector, an instance of the
+ *        controller class will be constructed and passed the element in
+ *        order to upgrade it.
+ */
+function upgradeElements(root, controllers) {
+  Object.keys(controllers).forEach(function (selector) {
+    var elements = Array.from(root.querySelectorAll(selector));
```

Comment:
`Array.from` doesn't appear to be supported at all in IE. Are we sure that's ok?